test/scenarios.py

```diff
@@ -12317,4 +12317,190 @@
+                                    1048576,
+                                ),
+                            },
+                        )
+                    ),
+                    action.Wait(event.Start(0, FILES["testfile-small"].id)),
+                    action.Wait(
+                        event.FinishFileUploaded(
+                            0,
+                            FILES["testfile-small"].id,
+                        )
+                    ),
+                    action.ExpectCancel([0], True),
+                    action.NoEvent(),
+                    action.Stop(),
+                ]
+            ),
+            "DROP_PEER_STIMPY": ActionList(
+                [
+                    action.Start(
+                        "DROP_PEER_STIMPY",
+                        checksum_events_size_threshold=0,
+                        checksum_events_granularity=256 * 1024 * 2,
+                    ),
+                    action.Wait(
+                        event.Receive(
+                            0,
+                            "DROP_PEER_REN",
+                            {
+                                event.File(
+                                    FILES["testfile-small"].id,
+                                    "testfile-small",
+                                    1048576,
+                                ),
+                            },
+                        )
+                    ),
+                    action.Download(
+                        0,
+                        FILES["testfile-small"].id,
+                        "/tmp/received/53-2",
+                    ),
+                    action.Wait(event.Pending(0, FILES["testfile-small"].id)),
+                    action.Wait(event.Start(0, FILES["testfile-small"].id)),
+                    action.Wait(
+                        event.ChecksumStarted(0, FILES["testfile-small"].id, 1048576)
+                    ),
+                    action.Wait(
+                        event.ChecksumProgress(
+                            0, FILES["testfile-small"].id, checksummed_bytes=524288
+                        )
+                    ),
+                    action.Wait(event.ChecksumFinished(0, FILES["testfile-small"].id)),
+                    action.Wait(
+                        event.FinishFileDownloaded(
+                            0,
+                            FILES["testfile-small"].id,
+                            "/tmp/received/53-2/testfile-small",
+                        )
+                    ),
+                    action.CheckDownloadedFiles(
+                        [
+                            action.File("/tmp/received/53-2/testfile-small", 1048576),
+                        ],
+                    ),
+                    action.CancelTransferRequest([0]),
+                    action.ExpectCancel([0], False),
+                    action.NoEvent(),
+                    action.Stop(),
+                ]
+            ),
+        },
+    ),
```

Comment:
Good testcases. Checksumming happens when receiving file as well, meaning if A sends file to B and there's a disconnect in between then B will calculate checksum once it tries to resume the file download, so this testcase could potentially also be added.(you could model after 21-1) but it's not strictly necessary, this is already good enough :)

Reply:
I think it's redundant because it's the same function. Slightly different flow, but the same checksum function. Or am I missing something?

Reply:
Same functionality just a different flow, yep. It's redundant in terms of functionality though. It's good enough without it, but it should be very easy to add regardless

Reply:
Ok, on it